src/main/java/org/dataone/hashstore/hashstoreconverter/FileHashStoreLinks.java

```diff
@@ -0,0 +1,233 @@
+ * operation.
+ */
+public class FileHashStoreLinks extends FileHashStore {
+
+    private static final Log logFileHashStoreLinks = LogFactory.getLog(FileHashStore.class);
+    private final int DIRECTORY_DEPTH;
+    private final int DIRECTORY_WIDTH;
+    private final String OBJECT_STORE_ALGORITHM;
+    private final Path OBJECT_STORE_DIRECTORY;
+
+    /**
+     * Constructor for FireHashStoreLinks. HashStore properties are required.
+     *
+     * @param hashstoreProperties Properties object with the following keys: storePath, storeDepth,
+     *                            storeWidth, storeAlgorithm, storeMetadataNamespace
+     * @throws IllegalArgumentException If there is an issue with one of the properties supplied
+     * @throws IOException              An issue with reading or writing a hashstore.yaml
+     *                                  configuration file
+     * @throws NoSuchAlgorithmException If an algorithm in the properties is not supported
+     */
+    public FileHashStoreLinks(Properties hashstoreProperties)
+        throws IllegalArgumentException, IOException, NoSuchAlgorithmException {
+        super(hashstoreProperties);
+        // If configuration matches, set FileHashStoreLinks private variables
+        Path storePath =
+            Paths.get(hashstoreProperties.getProperty(HashStoreProperties.storePath.name()));
+        int storeDepth = Integer.parseInt(
+            hashstoreProperties.getProperty(HashStoreProperties.storeDepth.name()));
+        int storeWidth = Integer.parseInt(
+            hashstoreProperties.getProperty(HashStoreProperties.storeWidth.name()));
+        String storeAlgorithm =
+            hashstoreProperties.getProperty(HashStoreProperties.storeAlgorithm.name());
+        DIRECTORY_DEPTH = storeDepth;
+        DIRECTORY_WIDTH = storeWidth;
+        OBJECT_STORE_ALGORITHM = storeAlgorithm;
+        OBJECT_STORE_DIRECTORY = storePath.resolve("objects");
+        logFileHashStoreLinks.info("FileHashStoreLinks initialized");
+    }
+
+    /**
+     * Store a hard link to HashStore from an existing data object in the filesystem.
+     *
+     * @param filePath          Path to the source file which a hard link will be created for
+     * @param fileStream        Stream to the source file to calculate checksums for
+     * @param pid               Persistent or authority-based identifier for tagging
+     * @param checksum          Value of checksum
+     * @param checksumAlgorithm Ex. "SHA-256"
+     * @return ObjectMetadata encapsulating information about the data file
+     * @throws NoSuchAlgorithmException Issue with one of the hashing algorithms to calculate
+     * @throws IOException              An issue with reading from the given file stream
+     * @throws InterruptedException     Sync issue when tagging pid and cid
+     */
+    public ObjectMetadata storeHardLink(
+        Path filePath, InputStream fileStream, String pid, String checksum,
```

Comment:
The filePath and fileStream parameters seem like redundant.  The fileStream must be read from the filePath. We can move it to a local variable. The redundant parameters may introduce an error - the fileStream may not be read from the file path.